more_itertools/more.py

```diff
@@ -181,7 +188,13 @@ def __getitem__(self, index):
         if isinstance(index, slice):
             return self._get_slice(index)
 
-        return self._get_slice(slice(index, index + 1, None))[0]
+        cache_len = len(self._cache)
+        if index < 0:
```

Comment:
Ah, yes. I had those two things together originally, but pulled them apart in 70c7767, since a `None` slice boundary has different meaning depending on the sign of step.